pwa/src/templates/templateParts/mappingForm/MappingFormTemplate.tsx

```diff
@@ -51,6 +62,19 @@ export const MappingFormTemplate: React.FC<MappingFormTemplateProps> = ({ mappin
     ref.current.scrollIntoView({ behavior: "smooth", block: "nearest", inline: "start" });
   };
 
+  const handleSwitchEditor = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, name: "mapping" | "unset" | "cast") => {
+    e.stopPropagation();
+
+    switch (name) {
+      case "mapping":
+        setEditMappingInEditor(!editMappingInEditor);
```

Comment:
Whenever toggling a state, we should hook into the current state using the callback function. Right now, this could break during the React lifecycle. 

e.g.
`setEditMappingInEditor((currentMappingInEditor) => !currentMappingInEditor);`

This way you can be 100% sure you're inverting the right value.